src/storage-controller/src/lib.rs

```diff
@@ -1129,7 +1126,85 @@ where
                 .expect("poisoned")
                 .insert(id, statistics::StatsInitState(BTreeMap::new()));
 
-            client.send(StorageCommand::CreateSinks(vec![cmd]));
+            client.send(StorageCommand::RunSinks(vec![cmd]));
+        }
+        Ok(())
+    }
+
+    /// Create the sinks described by the `ExportDescription`.
+    async fn update_export_connection(
+        &mut self,
+        exports: BTreeMap<GlobalId, StorageSinkConnection>,
+    ) -> Result<(), StorageError> {
+        for (id, connection) in exports {
+            let export = self.export_mut(id).expect("export exists");
+            let current_sink = export.description.sink.clone();
+            export.description.sink.connection = connection;
```

Comment:
this should not happen until AFTER all the `?`'s in this function, unless we guarantee that any error returned from this function is not recoverable

Reply:
Sure thing––reordered this to do that. None of these errors are recoverable, but better safe than sorry.